daemon/container_operations.go

```diff
@@ -649,7 +649,10 @@ func cleanOperationalData(es *network.EndpointSettings) {
 }
 
 func (daemon *Daemon) updateNetworkConfig(container *container.Container, n *libnetwork.Network, endpointConfig *networktypes.EndpointSettings, updateSettings bool) error {
-	if containertypes.NetworkMode(n.Name()).IsUserDefined() {
+	// Set up DNS names for a user defined network, and for the default 'nat'
+	// network on Windows (IsBridge() returns true for nat).
+	if containertypes.NetworkMode(n.Name()).IsUserDefined() ||
+		(serviceDiscoveryOnDefaultNetwork() && containertypes.NetworkMode(n.Name()).IsBridge()) {
```

Comment:
Could we ever hit the `default` network here (so not "nat")? https://github.com/moby/moby/blob/a60546b084eef87910345876e32c3db46e7c3bbc/api/types/container/hostconfig.go#L140-L143

Wondering if we should include a link to the github discussion here for future reference 🤔

Reply:
I tried to get "default" as far as this, but couldn't - so, decided not to add it. I couldn't see any problem with adding it just-in-case but, because I couldn't test it, I left it out and made a change targeted as specifically as possible at the issue being fixed. (Maybe libnetwork has a surprise to spring with "default", it usually does!)

For the github link ... I think the comment probably covers it, and git-blame is there if needed. But, could do?

As noted in the description, we need a follow-up change to clean this up properly - probably by adding a better predicate and looking at where else it should be used (and what the consequences would be). For example, I think the test in [setupPathsAndSandboxOptions](https://github.com/moby/moby/blob/97478c99f8dfe5f59d80932efb17bb699f4d7030/daemon/container_operations_unix.go#L427-L469) for whether to use systemd's resolv.conf also uses `IsUserDefined` as a proxy for "has internal resolver", so it suffers a similar problem... swarm-scoped networks, which do have an internal resolver, end up using systemd's underlying nameservers instead of `127.0.0.53`... I think they miss the `IsUserDefined()` case by having the name `default` when it matters. I think that's probably an oversight but, as-ever, changing it might break things.

Reply:
Yeah, maybe the `default` is not an issue (but I'm definitely not a fan of that magic value 😂 https://github.com/moby/moby/issues/19421)

Adding a link is not a real show-stopper for me; was mostly considering if we wanted to have a crumble-path to find back the discussion around this (basically; this is really a bit of a stop-gap, pending a more solid definition of all of this).